server/src/main/java/org/opensearch/index/engine/EngineConfig.java

```diff
@@ -460,6 +465,16 @@ public LongSupplier getPrimaryTermSupplier() {
         return primaryTermSupplier;
     }
 
+    /**
+     * Returns if this replica should be wired as a read only.
+     * This is used for Segment Replication where the engine implementation used is dependent on
+     * if the shard is a primary/replica.
+     * @return true if this engine should be wired as read only.
+     */
+    public boolean isReadOnlyReplica() {
+        return isReadOnlyReplica;
```

Comment:
Check if segrep feature flag is enabled?

Reply:
added.  It's also checked when config is created in IndexShard.